integration/test_aggregation_integration.py

```diff
@@ -4085,3 +4085,57 @@ def test_aggregate_and_query_by_key_with_holes(setup_teardown_test):
     assert (
         actual == expected_results
     ), f"actual did not match expected. \n actual: {actual} \n expected: {expected_results}"
+
+
+def test_float_format(setup_teardown_test):
+    t0 = pd.Timestamp(test_base_time)
+    floats_array = [1.5089695129344164e-05, 1.07174329054775652e05, 1]
```

Comment:
not sure why, but when I tried this test without the fix it still passed.. do you understand why?
btw, changing floats_array to `floats_array = [1.5089695129344164e05,  1, 1.07174329054775652e-05]` does fail on the old code.

Reply:
I know why - because the fault happens on the last element of the floats_array. When '1' the old code handles it well. Will fix the test

Reply:
Test still does not fail without the fix.

Reply:
will fix this on top of alex commits

Reply:
Just make your own branch and PR? It's free :)